src/core/service.c

```diff
@@ -2695,6 +2698,13 @@ static void service_sigchld_event(Unit *u, pid_t pid, int code, int status) {
                                 /* Fall through */
 
                         case SERVICE_RUNNING:
+                                if (s->type == SERVICE_NOTIFY) {
+                                        /* Only enter running through a notification */
+                                        if (f != SERVICE_SUCCESS)
+                                                service_enter_signal(s, SERVICE_FINAL_SIGTERM, f);
+                                        break;
+                                }
+
                                 service_enter_running(s, f);
```

Comment:
Checking for notifications is likely not possible. Only the last notification is remembered and `READY=1` has a second meaning in case of a reload. For tracking whether an non-reload READY notification has been received, we probably have to add a state flag.